src/main/java/org/gradle/crypto/checksum/Checksum.java

```diff
@@ -87,49 +102,55 @@ public void setOutputDir(File outputDir) {
         this.outputDir = outputDir;
     }
 
+
     @TaskAction
     public void generateChecksumFiles(IncrementalTaskInputs inputs) throws IOException {
```

Comment:
I suppose you can't migrate to use `InputChanges` since you want the plugin to work on versions prior to Gradle 5.4? If you can migrate it I suggest you do, given that we are going to start nagging about this soon.

Reply:
I have already restricted the compatibility to 4+ and I'd like not to go further.
I keep this in mind for the next iteration though 👍

Reply:
@jprinet Gradle 4.0 was released in June 14, 2017. If this were my community plugin, I would not be overly cautious with supporting ancient Gradle versions.

Also, if the motivation is to add configuration cache support, people still being on Gradle 4 would not benefit from it anyway.

I would make Gradle 6 the min requirement and refer users of older versions of Gradle to one of the current versions of this plugin.

Just my 2 cents.

Reply:
Yep, this was an opened discussion we had in the solution's sync yesterday, what's better between complex version dependent logic and a compatibility matrix.

This PR aims to provide build cache as well, that's why version 4.0 could still make sense.

Given your suggestions all, I'll stick to the deprecated `IncrementalTaskInputs` for this PR and will open a new one using `InputChanges` and getting rid of the conditional logic, bumping the version to next major.

Reply:
In the end, I believe it is the call of @eskatos but personally, I find it better to support recent versions of Gradle and their new functionality than be constrained by old Gradle versions. In particular given this is not a core or commercial plugin.

Doing this via different PRs, is surely a good idea, @jprinet 👍🏻

Reply:
I assume you meant @wolfs here, not me.